src/modules/authorization/sagas.ts

```diff
@@ -155,6 +183,91 @@ export function createAuthorizationSaga() {
     }
   }
 
+  function* handleAuthorizationFlowRequest(
+    action: AuthorizationFlowRequestAction
+  ) {
+    const { authorizationAction, authorization, allowance } = action.payload
+    const isRevoke = authorizationAction === AuthorizationAction.REVOKE
+    const tokenRequest = isRevoke
+      ? revokeTokenRequest(authorization)
+      : grantTokenRequest(authorization)
+    const TOKEN_SUCCESS = isRevoke ? REVOKE_TOKEN_SUCCESS : GRANT_TOKEN_SUCCESS
+    const TOKEN_FAILURE = isRevoke ? REVOKE_TOKEN_FAILURE : GRANT_TOKEN_FAILURE
+
+    yield put(tokenRequest)
+
+    try {
+      const {
+        success,
+        failure
+      }: {
+        success: RevokeTokenSuccessAction | GrantTokenSuccessAction | undefined
+        failure: RevokeTokenFailureAction | RevokeTokenFailureAction | undefined
+      } = yield race({
+        success: take(TOKEN_SUCCESS),
+        failure: take(TOKEN_FAILURE)
+      })
+      if (failure) {
+        throw new Error(failure.payload.error)
+      }
+      const txHash = getTransactionFromAction(success!).hash
```

Comment:
could you please modify the non-null assertion?

```suggestion
      const txHash = success && getTransactionFromAction(success).hash
```

Reply:
Done! I added `success as TokenSuccessAction` because if i did the `success && ` then i would have to solve txHash being undefined also which is a case that can't happen as success or failure should have a value and if failure has it will throw an exception so when reaching the code success has to be different than undefined